pkg/kubectl/cmd/get.go

```diff
@@ -110,19 +116,23 @@ func RunGet(f *cmdutil.Factory, out io.Writer, cmd *cobra.Command, args []string
 	if isWatch || isWatchOnly {
 		r := resource.NewBuilder(mapper, typer, f.ClientMapperForCommand()).
 			NamespaceParam(cmdNamespace).DefaultNamespace().AllNamespaces(allNamespaces).
+			FilenameParam(enforceNamespace, filenames...).
 			SelectorParam(selector).
 			ResourceTypeOrNameArgs(true, args...).
 			SingleResourceType().
+			Latest().
 			Do()
 		if err != nil {
 			return err
 		}
 
-		mapping, err := r.ResourceMapping()
+		infos, err := r.Infos()
 		if err != nil {
 			return err
 		}
 
+		info := infos[0]
```

Comment:
Since it expects a single resource, we can first make sure length doesn't exceed `1`, like what we did here:  https://github.com/kubernetes/kubernetes/blob/master/pkg/kubectl/cmd/expose.go#L105